### PR TITLE
oob performance improvement for cpu DDP

### PR DESCRIPTION
**Pull request by sywangyi**
Signed-off-by: Wang, Yi A <yi.a.wang@intel.com>

# What does this PR do?

oob performance improvement for cpu DDP, currently if no OMP_NUM_THREADS/MKL_NUM_THREADS is set, num_cpu_threads_per_process is set to 1. very slow performance.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

- trainer: @sgugger

### CODE REVIEW

**Comment by sywangyi** [comment]
@yao-matrix @sgugger @liangan1 please help review it

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -1335,6 +1335,17 @@ def _setup_devices(self) -> "torch.device":
                             "Looks like distributed multinode run but MASTER_ADDR env not set, "
                             "please try exporting rank 0's hostname as MASTER_ADDR"
                         )
+                if torch.get_num_threads() == 1 and get_int_from_env(["OMP_NUM_THREADS", "MKL_NUM_THREADS"], 0) == 0:
+                    import psutil
```
Comment (sgugger) [inline src/transformers/training_args.py:1339]: `psutil` is not a hard dependency of Transformers, so you should add a `is_psutil_available` to the test above.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this, left a comment.

**Comment by LysandreJik** [comment]
Hey @sywangyi, thanks for your PR! Sylvain is currently off for a few weeks, we'll merge this PR once he's back.

Thanks for your contribution!

**Review by sgugger — APPROVED** [review]
Thanks again for your PR!

**Pull request merged**